pyanaconda/modules/boss/boss.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/python3
+
+from pyanaconda.dbus import dbus_constants
+from pyanaconda.dbus.interface import dbus_interface
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.base import BaseModule
+
+from .module_manager import ModuleManager  # pylint: disable=relative-beyond-top-level
+
+from pyanaconda import anaconda_logging
+log = anaconda_logging.get_dbus_module_logger(__name__)
+
+@dbus_interface(dbus_constants.DBUS_BOSS_NAME)
+class Boss(BaseModule):
+
+    def __init__(self):
+        super().__init__()
+        self._dbus_name = dbus_constants.DBUS_BOSS_NAME
+        self._module_manager = ModuleManager(bus=self._bus)
+
+    def run(self):
+        log.info("looking for addons")
+        self._module_manager.find_addons()
+        log.info("starting all modules")
+        self._module_manager.start_modules()
+        log.info("publishing itself & starting mainloop")
+        self.publish_module()
```

Comment:
Also, shouldn't be the boss published first? Maybe starting of modules should be "planned" to run in a loop.

Reply:
If it could take longer time then it definitely should run in a loop.

Reply:
That's a good point - thinking about it the order of the initial startup could be quite important, even though we might find out the details only once we have more stuff on DBUS.
So lets just:
1) publish bos
2) schedule module startup on idle_add (so that modules can call Boss during startup ?)
3) start main loop
And see where this gets us. :)